### PR TITLE
fix ens contenthash resolution

### PR DESCRIPTION
**Pull request by yenda**
make ens resolution of contenthash more robust.

### CODE REVIEW

**Comment by flexsurfer** [comment]
thanks @yenda ! could you please add this case to tests `0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000026e301017012206e70f0f72e2bcd0ad69bb2b15ee9e0f4f88693f6b2f485e48699e7d6f8dbd62a0000000000000000000000000000000000000000000000000000`

File: src/status_im/ethereum/ens.cljs [hunk 1]
```diff
@@ -105,10 +106,21 @@
...
-            [cid & hash-and-zeros] (string/split hash "1b20")
-            hash (str "0x" cid "1b20" (subs (apply str hash-and-zeros) 0 64))]
-        (cb hash)))}))
+      (let [;; ignore hex prefix
+            [_ raw-hash-rest] (split-at 2 raw-hash)
+            ;; the first field gives us the length of the next one in hex and has
+            ;; a length of 32 bytes
+            ;; 1 byte is 2 chars here
+            [next-field-length-hex raw-hash-rest] (split-at 64 raw-hash-rest)
+            next-field-length (* (abi-spec/hex-to-number (apply str next-field-length-hex)) 2)
```
Comment (rasom) [inline src/status_im/ethereum/ens.cljs]: `apply str` -> `string/join`?

**Review by rasom — APPROVED** [review]

**Comment by flexsurfer** [comment]
@yenda test fails 
```
2020-03-31T20:20:00.057Z] FAIL in (contenthash-decode) (cljs/test.cljs:476:9)

[2020-03-31T20:20:00.057Z] decoding a valid swarm hash

[2020-03-31T20:20:00.057Z] expected: (= (contenthash/decode "0xe40101fa011b20d1de9994b4d039f6548d191eb26786769f580809256b4685ef316805265ea162") {:namespace :swarm, :hash "d1de9994b4d039f6548d191eb26786769f580809256b4685ef316805265ea162"})

[2020-03-31T20:20:00.058Z]   actual: (not (= nil {:namespace :swarm, :hash "d1de9994b4d039f6548d191eb26786769f580809256b4685ef316805265ea162"}))
```

File: src/status_im/utils/contenthash.cljs [hunk 2]
```diff
@@ -32,12 +32,12 @@
...
       ;; content type can be 2 or 4 bytes
-      ;; we expect 1b20 (hash algorithm and hash length 64)
+      ;; we expect 1b20 (hash algorithm keccak256 and hash length 64)
       ;; before the hash so we split the contenthash there
       (when-let [hash (second (string/split hex "1b20"))]
-        (when (= 0x20 (count hash)))
-        {:namespace :swarm
-         :hash hash})
+        (when (= 0x20 (count hash))
+          {:namespace :swarm
```
Comment (flexsurfer) [inline src/status_im/utils/contenthash.cljs]: probably test fails because of this

**Pull request merged**